transport/wsn/src/main/java/org/kaazing/gateway/transport/wsn/WsnAcceptor.java

```diff
@@ -1114,22 +1112,16 @@ private void doUpgrade(final HttpAcceptSession session) throws Exception {
 
                     // Send out 302 HTTP Balance for WSN and WSX 
                     Object balancerKeys = session.getParent().getAttribute(BALANCEES_KEY);
-                    if (!"x-kaazing-handshake".equals(wsProtocol0) && wsVersion == HYBI_13 && balancerKeys != null) {
-                        SocketAddress parentLocalAddress = session.getParent().getLocalAddress();
-                        boolean isWsx = parentLocalAddress instanceof ResourceAddress
-                                && ((ResourceAddress) parentLocalAddress).getOption(CODEC_REQUIRED);
-    
-                        if ((WSN_302_REDIRECT.isEnabled(configuration) && !isWsx)
-                                || (isWsx && WSX_302_REDIRECT.isEnabled(configuration))) {
-                            assert balancerKeys instanceof List;
-                            List<String> availableBalanceeURIs = (List<String>) balancerKeys;
-                            String balanceURI = availableBalanceeURIs.get((int) (Math.random() * availableBalanceeURIs.size()));
-                            balanceURI = URLUtils.modifyURIScheme(URI.create(balanceURI), "http").toString();
-                            session.setStatus(HttpStatus.REDIRECT_FOUND);
-                            session.setWriteHeader("location", balanceURI);
-                            session.close(false);
-                            break;
-                        }
+                    if (!"x-kaazing-handshake".equals(wsProtocol0) && wsVersion == HYBI_13 && balancerKeys != null &&
+                            !"Y".equals(session.getParameter(".kl"))) {
```

Comment:
This looks good to me, thoughI'm not sure if the .kl is going to be presented properly in the wsx cases (I think the layering might remove it :-().  If we add the tests above, and they prove me wrong, then this works.  Otherwise, we will need some more logic